lib/problem2.rb

```diff
@@ -0,0 +1,17 @@
+def problem2(values)
+  raise ArgumentError, 'not enough elements for operation' if values.count < 2
+
+  first = nil
+  second = nil
+
+  values.each do |value|
+    if value > first.to_i then
```

Comment:
I guess you are doing `.to_i` because you init values with `nil` the downside is that you will to an unnecessary conversion when a value is populated already. Like haskell or any other modern language ruby has constants for max and min values, can you rewrite this code to make a little more efficient?

Reply:
Was not able to find the constants you where referring to, all I found where user defined constants in stackoverflow examples (or maybe they where taken from ruby source code). And those constants where defined for Fixnum, not Integer, which also contains Bignums.

Reply:
Indeed the MAX and MIN constant have not been yet released, there [is a way](https://gist.github.com/pithyless/9738125) to get those constants now.

Looking at your solution you actually don't need the constants any, do you? You can simply assign the variables initially with the first 2 values of the array, of course you need to compare them first.

One intresting ruby like solution will be something like

```ruby
first, second = values.first(2)
first, second = second, first if second > first
```

What do you think, can you fill in the rest of the code?